pkg/e2e/operators/cloudingress/installed.go

```diff
@@ -2,32 +2,98 @@ package cloudingress
 
 import (
 	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
 
 	"github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-
 	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/spf13/viper"
 
+	"github.com/openshift/osde2e/pkg/common/config"
+	appsv1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // tests
+
 var _ = ginkgo.Describe(CloudIngressTestName, func() {
+
+	var operatorName = "cloud-ingress-operator"
+	var operatorNamespace string = "cloud-ingress-operator"
+	var defaultDesiredReplicas int32 = 1
+
 	h := helper.New()
 
-	ginkgo.It("is a placeholder", func() {
-		_, _ = h.Kube().CoreV1().Pods(CloudIngressNamespace).Get(context.TODO(), "something", metav1.GetOptions{})
+	checkDeployment(h, operatorNamespace, operatorName, defaultDesiredReplicas)
+})
 
-		tests := []struct {
-			Name string
-		}{}
+func checkDeployment(h *helper.H, namespace string, name string, defaultDesiredReplicas int32) {
+	// Check that the operator deployment exists in the operator namespace
+	ginkgo.Context("deployment", func() {
+		ginkgo.It("should exist", func() {
+			deployment, err := pollDeployment(h, namespace, name)
+			Expect(err).ToNot(HaveOccurred(), "failed fetching deployment")
+			Expect(deployment).NotTo(BeNil(), "deployment is nil")
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+		ginkgo.It("should have all desired replicas ready", func() {
+			deployment, err := pollDeployment(h, namespace, name)
+			Expect(err).ToNot(HaveOccurred(), "failed fetching deployment")
 
-		for _, test := range tests {
-			Expect(test.Name).To(Equal(""))
-		}
+			readyReplicas := deployment.Status.ReadyReplicas
+			desiredReplicas := deployment.Status.Replicas
+
+			// The desired replicas should match the default installed replica count
+			Expect(desiredReplicas).To(BeNumerically("==", defaultDesiredReplicas), "The deployment desired replicas should not drift from the default 1.")
+
+			// Desired replica count should match ready replica count
+			Expect(readyReplicas).To(BeNumerically("==", desiredReplicas), "All desired replicas should be ready.")
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
 	})
-})
+}
+
+func pollDeployment(h *helper.H, namespace, deploymentName string) (*appsv1.Deployment, error) {
+	// pollDeployment polls for a deployment with a timeout
+	// to handle the case when a new cluster is up but the OLM has not yet
+	// finished deploying the operator
+
+	var err error
+	var deployment *appsv1.Deployment
+
+	// interval is the duration in seconds between polls
+	// values here for humans
+	interval := 5
 
-// utils
+	// convert time.Duration type
+	timeoutDuration := time.Duration(viper.GetFloat64(config.Tests.PollingTimeout)) * time.Minute
+	intervalDuration := time.Duration(interval) * time.Second
+
+	start := time.Now()
+
+Loop:
```

Comment:
I'm not familiar with this keyword - what's it do?

Reply:
It sets a reference point from which you can measure time elapsed. For example,  start := time.Now()  *something happens in between* end := time.Now()  time_elapsed:=start.Sub(end) More info [here](https://golang.org/pkg/time/) .

Reply:
Sorry, I tried to put this comment on the `Loop:` line.

I didn't realize that loops could be labeled and targeted by a `break` statement like this. Neat!